subsys/bluetooth/host/shell/bt.c

```diff
@@ -826,17 +821,33 @@ static void disconnected(struct bt_conn *conn, uint8_t reason)
 	bt_shell_print("Disconnected: %s (reason 0x%02x)", addr, reason);
 
 	if (default_conn == conn) {
+		struct bt_conn_info info;
+		enum bt_conn_type conn_type = BT_CONN_TYPE_LE;
+
+		if (IS_ENABLED(CONFIG_BT_CLASSIC)) {
+			conn_type |= BT_CONN_TYPE_BR;
+		}
+
+		bt_conn_get_info(conn, &info);
 		bt_conn_unref(default_conn);
 		default_conn = NULL;
 
 		/* If we are connected to other devices, set one of them as default */
-		bt_conn_foreach(BT_CONN_TYPE_LE, disconnected_set_new_default_conn_cb, NULL);
+		bt_conn_foreach(info.type, disconnected_set_new_default_conn_cb, NULL);
```

Comment:
Wouldn't it be more accurate like this?:

```suggestion
		bt_conn_foreach(BT_CONN_TYPE_LE | BT_CONN_TYPE_BR, disconnected_set_new_default_conn_cb, NULL);
```

I believe it's more accurate because we don't really care about what the type of the disconnected connection is for the purpose of selecting a different connection.

I don't know what the difference between BT_CONN_TYPE_BR and BT_CONN_TYPE_SCO is, so it may be SCO is correct instead.

Reply:
Hi alwa-nordic, as the comments from Thalley, he expects to set a new default conn with the same type first.
Consider a requirement: If there are 4 connections: 2 LE and 2 BR/EDR, and the disconnected one is a BR/EDR, and we want to select the other BR/EDR connection.

SCO disconnection will not reach this callback and will use its own callback.